Userland/Libraries/LibJS/Runtime/Temporal/ISO8601.cpp

```diff
@@ -824,31 +824,76 @@ bool ISO8601Parser::parse_time_zone_utc_offset_name()
     return true;
 }
 
-// https://tc39.es/proposal-temporal/#prod-TimeZoneIANAName
-bool ISO8601Parser::parse_time_zone_iana_name()
+// https://tc39.es/proposal-temporal/#prod-TZLeadingChar
+bool ISO8601Parser::parse_tz_leading_char()
 {
     // TZLeadingChar :
     //     Alpha
     //     .
     //     _
+    if (m_state.lexer.next_is(is_ascii_alphanumeric)) {
+        m_state.lexer.consume();
+        return true;
+    }
+    return m_state.lexer.consume_specific('.')
+        || m_state.lexer.consume_specific('_');
+}
+
+// https://tc39.es/proposal-temporal/#prod-TZChar
+bool ISO8601Parser::parse_tz_char()
+{
     // TZChar :
     //     Alpha
     //     .
     //     -
     //     _
+    return parse_tz_leading_char()
```

Comment:
A comment might be nice, since this might break if `TZLeadingChar` is changed in the future.

Reply:
Eh, just duplicated the code :)